### ISSUE TITLE
Is the corrected AIC correct?

### ISSUE THREAD

**Issue by pjgorski** [TaddyLab/gamlr#17]
The current version of the corrected AIC has the correction that does not vanish with increasing $n$. It is reported that the formula follows Hurvich and Tsai, 1989. 

In the paper, the formula is as follows:
![image](https://github.com/TaddyLab/gamlr/assets/16017475/4ec87db1-1a76-4f43-b7d1-b9cdcc8614ad)

So the term $k*d*n/(n-d-1)$, which is equal to $\dfrac{2(n-m)n}{m-1}$, is not equal to the term from the paper.

**Comment by mataddy** [comment]
yep it’s correct.  d = n-m in their special case.  read up on the degrees of freedom section.On Nov 28, 2023, at 4:08 AM, pjgorski ***@***.***> wrote:﻿
The current version of the corrected AIC has the correction that does not vanish with increasing $n$. It is reported that the formula follows Hurvich and Tsai, 1989.
In the paper, the formula is as follows:
image.png (view on web)
So the term $kdn/(n-d-1)$, which is equal to $\dfrac{2(n-m)n}{m-1}$, is not equal to the term from the paper.

**Comment by pjgorski** [comment]
You're right. Your `AICc` is correct. 

However, what you wrote, $d=n-m$, is incorrect for your implementation. $n-m$ is degrees of freedom for LR models. $d$ in your implementation is degrees of freedom of `logLik` function, which is something different (I think it is the number of parameters that would be estimated when doing MLE). That is why the documentation of `AICc` was misleading to me when I read it. 

(I didn't understand what you meant by "read up on the degrees of freedom section". I didn't see such a section in the mentioned paper.)

As a sidenote, I will point that `AICc` implementation from [qpcR](https://www.rdocumentation.org/packages/qpcR/versions/1.4-1/topics/AICc) library is incorrect. Their implemented function follows the function from [wikipedia](https://en.wikipedia.org/wiki/Akaike_information_criterion#AICc) but they just use the number of coefficients from LR and they do not include the variance estimate one.
